internal/bridge/crunchybridgecluster/crunchybridgecluster_controller_test.go

```diff
@@ -0,0 +1,156 @@
+//go:build envtest
+// +build envtest
+
+/*
+ Copyright 2021 - 2024 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package crunchybridgecluster
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crunchydata/postgres-operator/internal/testing/require"
+)
+
+func TestGetSecretKeys(t *testing.T) {
+	ctx := context.Background()
+	_, tClient := setupKubernetes(t)
+	require.ParallelCapacity(t, 0)
+
+	reconciler := &CrunchyBridgeClusterReconciler{Client: tClient}
+
+	ns := setupNamespace(t, tClient).Name
+	cluster := testCluster()
+	cluster.Namespace = ns
+
+	t.Run("NoSecret", func(t *testing.T) {
+		apiKey, team, err := reconciler.GetSecretKeys(ctx, cluster)
+		assert.Check(t, apiKey == "")
+		assert.Check(t, team == "")
+		assert.ErrorContains(t, err, "secrets \"crunchy-bridge-api-key\" not found")
+	})
+
+	t.Run("SecretMissingApiKey", func(t *testing.T) {
+		cluster.Spec.Secret = "secret-missing-api-key"
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "secret-missing-api-key",
+				Namespace: ns,
+			},
+			Data: map[string][]byte{
+				"team": []byte(`jkl;`),
+			},
+		}
+		assert.NilError(t, tClient.Create(ctx, secret.DeepCopy()))
```

Comment:
Do these need to be `DeepCopy` of the secrets? I see this being done a few times in this test.

I would expect a `DeepCopy` if we are updating the value of the secret and checking contents before and after